xarray/core/indexing.py

```diff
@@ -1291,9 +1302,27 @@ def __getitem__(self, indexer):
 
         return result
 
-    def transpose(self, order):
+    def transpose(self, order) -> pd.Index:
         return self.array  # self.array should be always one-dimensional
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return ('%s(array=%r, dtype=%r)'
                 % (type(self).__name__, self.array, self.dtype))
+
+    def copy(self, deep: bool = True) -> 'PandasIndexAdapter':
+        # Not the same as just writing `self.array.copy(deep=deep)`, as
+        # shallow copies of the underlying numpy.ndarrays become deep ones
+        # upon pickling
+        if deep:
+            array = self.array.copy(deep=True)
+        else:
+            array = self.array
+        return PandasIndexAdapter(array, self._dtype)
+
+    def __copy__(self) -> 'PandasIndexAdapter':
```

Comment:
Do we need `__copy__`/`__deepcopy__` methods on PandasIndexAdapter? I'm not sure how these methods would get called, given that we already override `__copy__`/`__deepcopy__` on Dataset/DataArray to use the `.copy()` methods.

Reply:
Technically we don't. It's just for the sake of completeness. I can remove them if you prefer...

Reply:
I would slightly we rather remove them, just to minimize the amount of redundant/untested code.

Reply:
Done